### PR TITLE
Restructure project selection

### PR DESCRIPTION
**Pull request by oyamauchi**
## Description

The idea here is to move project selection out of the main form, and
have the user select a project after submitting. The headline of the
results section is now dynamic. The row-of-pills selector is gone;
it's now a dropdown menu on all layouts.

There's one major element of the new design that's not implemented
here yet: a no-project-selected state. (This implementation
automatically selects the alphabetically-first project that has
incentives available, which is usually battery storage.) The
no-selection state is pretty easily separated from the rest of the
work, so I'd rather do that anyway to keep this PR from getting too big.

There are no multiple-selection dropdowns anywhere anymore, so I
removed that code, but I did it in a separate commit for ease of
reviewing.

Some design notes:

- I changed the down-arrow on the right of the dropdown to purple, as
  the design says. This applies to the dropdowns in the form as well.

- To implement this, I made it so that the selected item in
  single-select dropdowns gets a checkmark to the left. (Previously,
  only multi-select dropdowns got that treatment.) This affects the
  dropdowns in the form as well. I actually prefer it this way ---
  better visual indication of which item is selected.

- The background color of the number badge (purple-100) is the same as
  the color of the active (i.e. hovered) row in the dropdown, so
  there's a gray border around the badge in the hovered/active state.

## Test Plan

- Submit the form, select a project other than the first one. Reload
  the page, submit the form again, make sure the previously selected
  project is restored.

- Put in a low income and submit the form. Select "cooking", which
  usually has only the IRA rebate card. Put in a high income and submit
  again. Make sure "cooking" is not selected (because it has zero
  incentives), and the first project that has nonzero incentives is
  selected.

In all of browsers [Chrome, Safari, Firefox, MobileSafari]:

- View the component on all layout widths; make sure it looks nice

- Interact with the component with the keyboard. (Nothing about its
  interaction has changed substantially, but just to make sure.)

- Interact with the component with VoiceOver, to make sure the number
  badge gets announced in an understandable way. (It'll say like
  `project, battery storage, 1 result` when you're on the main button.)

### CODE REVIEW

File: src/state-incentive-details.tsx [hunk 1]
```diff
@@ -463,83 +471,61 @@ type Props = {
...
   emailSubmitter,
 }) => {
   const { msg } = useTranslated();
 
-  // We're filtering out IRA rebates in favor of state-specific handling.
-  const allEligible = response.incentives
-    // Forward compatibility for when only eligible incentives are returned from
-    // the API, with no eligible flag.
-    .filter(i => i.eligible === true || i.eligible === undefined)
-    .filter(i => !isIRARebate(i));
```
Comment (oyamauchi) [inline src/state-incentive-details.tsx:479]: This filtering is all being done in the backend now; we're not losing any functionality here.

File: cypress/e2e/state-calculator.cy.ts [hunk 2]
```diff
@@ -88,54 +84,6 @@ describe('rewiring-america-state-calculator', () => {
       });
   });
 
-  it('shows an empty state if you are not eligible for any incentives for your chosen project', () => {
```
Comment (oyamauchi) [inline cypress/e2e/state-calculator.cy.ts:91]: I'm actually not sure it's even possible to get this state anymore? Need to think about it. If it's not, there's even more code I can delete.

**Review by oyamauchi — COMMENTED** [review]
Still TODO:

- Figure out whether the empty state is still possible; delete the code if not
- Save selected project to local storage

File: src/state-incentive-details.tsx [hunk 3]
```diff
@@ -426,29 +423,53 @@ const IncentiveGrid = forwardRef<HTMLDivElement, IncentiveGridProps>(
...
+      </>
+    );
   },
 );
 
+/**
+ * If you make a backward-incompatible change to the format of form value
+ * storage, increment the version in this key.
+ */
+const SELECTED_PROJECT_LOCAL_STORAGE_KEY = 'RA-calc-selected-project-v1';
```
Comment (matelau) [inline src/state-incentive-details.tsx:463]: nice 👍🏼

File: src/state-incentive-details.tsx [hunk 4]
```diff
@@ -463,83 +484,85 @@ type Props = {
...
-  const projectsWithIncentives = (
-    Object.entries(incentivesByProject) as [Project, Incentive[]][]
-  )
-    .filter(([, incentives]) => incentives.length > 0)
-    .map(([project]) => project);
-
-  const interestedProjects = selectedProjects.sort((a, b) =>
-    shortLabel(a, msg).localeCompare(shortLabel(b, msg)),
-  );
+  const countOfProjects = Object.values(incentivesByProject).filter(
```
Comment (matelau) [inline src/state-incentive-details.tsx:505]: image
image

🤔 So, the counts don't seem to be right between the heading and the bullets + projects, am I missing something?

Comment (oyamauchi) [reply src/state-incentive-details.tsx]: Nah, I keep messing up the logic of including the IRA rebates in the counts :/

**Review by matelau — APPROVED** [review]
👍🏼

**Pull request merged**